examples/examples_test.go

```diff
@@ -74,3 +75,19 @@ func providerPluginPathEnv() (string, error) {
 	}
 	return "PATH=" + os.Getenv("PATH") + pathSeparator + absPluginDir + pathSeparator + absTestPluginDir, nil
 }
+
+var envToUnset = [...]string{"AWS_SECRET_ACCESS_KEY", "AWS_ACCESS_KEY_ID", "AWS_SESSION_TOKEN"}
+
+// unsetAWSProfileEnv unsets the AWS_PROFILE and associated environment variables.
+// EKS token retrieval using the AWS_PROFILE seems to prefer the
+// the following variables over AWS_PROFILE so you end up with
+// authentication failures in the tests. So drop these environment
+// variables if set and reapply them after the test.
+func unsetAWSProfileEnv(t *testing.T) {
+	t.Helper()
+
+	for _, envVar := range envToUnset {
+		t.Setenv(envVar, "")
```

Comment:
This gets troublesome with parallel tests as you might know.

Reply:
Yes. I'm just mimicking prior art for the existing AWS Profile tests we had. The tests run in non-parallel mode (which `t.Setenv` ensures as it panics if the tests run in parallel). I don't think there is an trivial way to get rid of this.